app/assets/javascripts/admin/subscriptions/controllers/address_controller.js.coffee

```diff
@@ -1,11 +1,21 @@
 angular.module("admin.subscriptions").controller "AddressController", ($scope, $filter, StatusMessage, availableCountries) ->
   $scope.countries = availableCountries
+
   $scope.statesFor = (country_id) ->
     return [] unless country_id
-    $filter('filter')(availableCountries, {id: country_id})[0].states
+    country = $filter('filter')(availableCountries, {id: country_id}, true)[0]
```

Comment:
adding this ", true" is the fix to the issue, everything else in this PR is tests and refactoring.